plugin.video.nhlgcl/resources/lib/globals.py

```diff
@@ -0,0 +1,644 @@
+    except TypeError:
+        date = datetime(*(time.strptime(str(string), date_format)[0:6]))
+
+    return date
+
+
+def eastern_to_local(eastern_time):
+    utc = pytz.utc
+    eastern = pytz.timezone('US/Eastern')
+    eastern_time = eastern.localize(eastern_time)
+    # Convert it from Eastern to UTC
+    utc_time = eastern_time.astimezone(utc)
+    timestamp = calendar.timegm(utc_time.timetuple())
+    local_dt = datetime.fromtimestamp(timestamp)
+    # Convert it from UTC to local time
+    assert utc_time.resolution >= timedelta(microseconds=1)
+    return local_dt.replace(microsecond=utc_time.microsecond)
+
+
+def utc_to_local(utc_dt):
+    # get integer timestamp to avoid precision lost
+    timestamp = calendar.timegm(utc_dt.timetuple())
+    local_dt = datetime.fromtimestamp(timestamp)
+    assert utc_dt.resolution >= timedelta(microseconds=1)
+    return local_dt.replace(microsecond=utc_dt.microsecond)
+
+
+def local_to_eastern():
+    eastern = pytz.timezone('US/Eastern')
+    local_to_utc = datetime.now(pytz.timezone('UTC'))
+
+    eastern_hour = local_to_utc.astimezone(eastern).strftime('%H')
+    eastern_date = local_to_utc.astimezone(eastern)
+    # Don't switch to the current day until 4:01 AM est
+    if int(eastern_hour) < 3:
+        eastern_date = eastern_date - timedelta(days=1)
+
+    local_to_eastern = eastern_date.strftime('%Y-%m-%d')
+    return local_to_eastern
+
+
+def eastern_to_utc(eastern_time):
+    utc = pytz.utc
+    eastern = pytz.timezone('US/Eastern')
+    eastern_time = eastern.localize(eastern_time)
+    # Convert it from Eastern to UTC
+    utc_time = eastern_time.astimezone(utc)
+    return utc_time
+
+
+def get_params():
+    param = []
+    paramstring = sys.argv[2]
+    if len(paramstring) >= 2:
+        params = sys.argv[2]
+        cleanedparams = params.replace('?', '')
+        if (params[len(params) - 1] == '/'):
+            params = params[0:len(params) - 2]
+        pairsofparams = cleanedparams.split('&')
+        param = {}
+        for i in range(len(pairsofparams)):
+            splitparams = {}
+            splitparams = pairsofparams[i].split('=')
+            if (len(splitparams)) == 2:
+                param[splitparams[0]] = splitparams[1]
+
+    return param
+
+
+def add_stream(name, link_url, title, game_id, epg, icon=None, fanart=None, info=None, video_info=None, audio_info=None,
+               start_time=None):
+    ok = True
+    u = sys.argv[0] + "?url=" + urllib.quote_plus(link_url) + "&mode=" + str(104) + "&name=" + urllib.quote_plus(name) \
+        + "&game_id=" + urllib.quote_plus(str(game_id)) + "&epg=" + urllib.quote_plus(str(epg))
+
+    if start_time is not None:
+        u += '&start_time='+start_time
+
+    liz = xbmcgui.ListItem(name)
+    if icon is not None:
+        liz.setArt({'icon': icon, 'thumb': icon, 'fanart': fanart})
+    else:
+        liz.setArt({'icon': ICON, 'thumb': ICON, 'fanart': fanart})
+
+    if fanart is not None:
```

Comment:
this can safely be removed. Please check for `None` in the setArt call for the fanart variable